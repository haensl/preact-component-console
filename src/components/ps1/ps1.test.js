/* eslint-env node, jest */

import { h } from 'preact';
import render from 'preact-render-to-string';
import { DEFAULTS } from './';
import PS1 from './';

describe('PS1', () => {
  describe('empty', () => {
    let tree;

    beforeEach(() => {
      tree = render(<PS1 />);
    });

    it ('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });

    it('adds the default content', () => {
      expect(tree).toMatch(new RegExp(`>\\${DEFAULTS.content}<`));
    });
  });

  describe('content', () => {
    let tree;
    
    beforeEach(() => {
      tree = render(<PS1 content="$" />);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });

    it('adds the content', () => {
      expect(tree).toMatch(/\$/);
    })
  });

  describe('class', () => {
    let tree;

    beforeEach(() => {
      tree = render(<PS1 classes={{
        element: "testclass"
      }} />);
    });
    
    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the class property', () => {
      expect(tree).toMatch(/class="testclass"/);
    });

    it('adds the default content', () => {
      expect(tree).toMatch(new RegExp(`>\\${DEFAULTS.content}<`));
    });
  });
});
