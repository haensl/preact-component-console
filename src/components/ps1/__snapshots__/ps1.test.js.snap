// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PS1 class renders as expected 1`] = `"<span class=\\"testclass\\">$</span>"`;

exports[`PS1 content renders as expected 1`] = `"<span class=\\"console-ps1\\">$</span>"`;

exports[`PS1 empty renders as expected 1`] = `"<span class=\\"console-ps1\\">$</span>"`;
